src/arrayMethodJoin.js

```diff
@@ -5,8 +5,22 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let result = '';
+    const resultSeparator = separator !== undefined ? `${separator}` : ',';
```

Comment:
how about using `String` here? You can read about it [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String?retiredLocale=pl)